src/Microsoft.AspNet.Razor/Generator/Compiler/CodeBuilder/CSharp/Visitors/CSharpTagHelperFieldDeclarationVisitor.cs

```diff
@@ -48,7 +48,13 @@ protected override void Visit(TagHelperChunk chunk)
 
                     WritePrivateField(_tagHelperContext.RunnerTypeName,
                                       CSharpTagHelperCodeRenderer.RunnerVariableName,
-                                      "new " + _tagHelperContext.RunnerTypeName + "()");
+                                      "new "
+                                      + _tagHelperContext.RunnerTypeName
+                                      + "("
+                                      + _tagHelperContext.HtmlHelperPropertyName
+                                      + "."
+                                      + _tagHelperContext.HtmlEncoderPropertyName
+                                      + ")");
```

Comment:
:+1: , the `WritePrivateField` has 2 writes to it so its not all taht much duplication